src/celeritas/track/SortTracksAction.cc

```diff
@@ -35,9 +35,16 @@ std::string SortTracksAction::label() const
 void SortTracksAction::execute(ParamsHostCRef const&,
                                StateHostRef& states) const
 {
-    if (track_order_ == TrackOrder::partition_status)
+    switch (track_order_)
     {
-        detail::partition_tracks_by_status(states);
+        case TrackOrder::partition_status:
+            detail::partition_tracks_by_status(states);
+            break;
+        case TrackOrder::sort_step_limit_action:
+            detail::sort_tracks_by_action_id(states);
+            break;
+        default:
+            break;
```

Comment:
```suggestion
            CELER_ASSERT_UNREACHABLE();
```

Reply:
This can still be reached, e.g. the default case when we have `TrackOrder::unsorted` or `TrackOrder::shuffle`  this action is still called but shouldn't do anything. Maybe just specify all the enum variants?

Reply:
Oh good point. I didn't realize we were inserting the  action even in those cases; I think it would be better *not* to insert it rather than to have a nullop action?

Reply:
Anyway, for now if you're refactoring to a template, just make the enums explicit, and we can clean it up in a follow-on PR.

Reply:
> Oh good point. I didn't realize we were inserting the action even in those cases; I think it would be better _not_ to insert it rather than to have a nullop action?

We could do that but then we're sort of moving the logic out of the action? Also right now there is a bit of a redundancy between `ActionOrder` and `TrackOrder` for `SortTracksAction` as one sort of implies the other.